File: backend/models/task.model.js
// const mongoose = require("mongoose");
// const taskSchema = new mongoose.Schema({
//     taskTitle: { 
//         type: String, 
//         required: true 
//     },
//     taskDescription: { 
//         type: String, 
//         required: true 
//     },
//     taskStatus: { 
//         type: String, 
//         enum: ["Panding", "In Progress", "Completed"], 
//         default: "Panding" 
//     },
//     startDate: { 
//         type: Date, 
//         required: true 
//     },
//     endDate: { 
//         type: Date, 
//         required: true 
//     },
//     isDelete: { 
//         type: Boolean, 
//         default: false 
//     },
//     projectId: { 
//         type: mongoose.Schema.Types.ObjectId, 
//         ref: "Project", 
//         required: true 
//     }, 
//     assignedUsers: [
//         { 
//             type: mongoose.Schema.Types.ObjectId, 
//             ref: "User" 
//         }
//     ], 
//     createdBy: { 
//         type: mongoose.Schema.Types.ObjectId, 
//         ref: "User", 
//         required: true 
//     }, 
// }, { timestamps: true });

// module.exports = mongoose.model("Task", taskSchema);


const mongoose = require("mongoose");

const taskSchema = new mongoose.Schema({
    taskTitle: {
        type: String,
        required: true
    },
    taskDescription: {
        type: String,
        required: true
    },
    taskStatus: {
        type: String,
        enum: ["Pending", "In Progress", "Completed"],
        default: "Pending"
    },
    startDate: {
        type: Date,
        required: true
    },
    endDate: {
        type: Date,
        required: true
    },
    isDelete: {
        type: Boolean,
        default: false
    },
    projectId: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "Project",
        required: true
    },
    assignedUsers: [
        {
            type: mongoose.Schema.Types.ObjectId,
            ref: "User"
        }
    ],
    createdBy: {
        type: mongoose.Schema.Types.ObjectId,
        ref: "User",
        required: true
    }
}, { timestamps: true });

module.exports = mongoose.model("Task", taskSchema);

File: Todo_Backend/Models/TaskModel.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
namespace Todo_Backend.Models
{
    public class TaskModel
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; } = string.Empty;

        public string TaskTitle { get; set; } = string.Empty;

        public string TaskDescription { get; set; } = string.Empty;

        public string TaskStatus { get; set; } = "To Do";

        public DateTime StartDate { get; set; }

        public DateTime EndDate { get; set; }

        public bool IsDelete { get; set; } = false;

        [BsonRepresentation(BsonType.ObjectId)]
        public string ProjectId { get; set; } = string.Empty;

        [BsonRepresentation(BsonType.ObjectId)]
        public List<string> AssignedUsers { get; set; } = new();

        public string CreatedBy { get; set; } = string.Empty;
    }
}
